### ISSUE TITLE
Crash: Swapchain presentation failed: SuboptimalKHR

### ISSUE THREAD

**Issue by nullref** [shadps4-emu/shadPS4#1089]
Reported for revision 5a4d5b9f3138c930cb1d24bef86ec15018ab929e.

Running a game on Sway+amdgpu fails with this error:
```
[Debug] <Critical> vk_swapchain.cpp:operator():132: Assertion Failed!
Swapchain presentation failed: SuboptimalKHR
```

The relevant code is this:
```cxx
    auto result = instance.GetPresentQueue().presentKHR(present_info);
    if (result == vk::Result::eErrorOutOfDateKHR) {
        needs_recreation = true;
    } else {
        ASSERT_MSG(result == vk::Result::eSuccess, "Swapchain presentation failed: {}",
                   vk::to_string(result));
    }
```

Research suggests getting `SuboptimalKHR` from `presentKHR` could mean the window is no longer the same size as the buffer, and the solution could be to set `needs_recreation` to `true` as done for `eErrorOutOfDateKHR`.  Another solution could be to debounce for a while until the window hasn't been resized (assumed cause) for a little while, and then set `needs_recreation`.

I also tested this which worked for me:
```cxx
    } else if (result == vk::Result::eSuboptimalKHR) {
        // do nothing
```

**Comment by ElBread3** [comment]
Can report that this also happens when pressing F11 to exit full screen, particularly noticed this in Project Diva Future Tone, but this happens with other games as well!

**Comment by nullref** [comment]
Fixed by #1142.
